src/main/java/uk/gov/ons/ctp/common/state/BasicStateTransitionManager.java

```diff
@@ -38,7 +40,13 @@ public S transition(final S sourceState, final E event) throws CTPException {
       destinationState = outputMap.get(event);
     }
     if (destinationState == null) {
+      log.warn("No valid transition found from " + sourceState.getClass().getName()
```

Comment:
These print out the classname which is the generally the same for all states - as they are generic I can't think of anything other than toString() but this should work